core/state_transition.go

```diff
@@ -210,28 +217,34 @@ func (st *StateTransition) to() common.Address {
 
 func (st *StateTransition) useGas(amount uint64) error {
 	if st.gas < amount {
-		return vm.ErrOutOfGas
+		return ErrIntrinsicGas
```

Comment:
why change this error to ErrIntrinsicGas?

Reply:
because `TransitionDb()` in the reference implementation returns this error and it is expected in tests. from what I see, harmony implementation refactored the `TransitionDb()` a little bit, moving out some stuff to other methods like `useGas()` . thus I had to change `useGas()` to be consistent with the reference implementation. In general, the intrinsic gas for a transaction is the amount of gas that the transaction uses before any code runs.